components/Common/OramaSearchBar/search.ts

```diff
@@ -0,0 +1,26 @@
+import { search } from '@orama/orama';
+
+import type { Document, Orama, RawData, Schema } from '@orama/orama';
+import type { SearchResult } from '../../../types';
+
+export type DocumentMapper = (doc: Document) => SearchResult;
+
+export type OramaSearchBarProps = {
+  schema: Schema;
+  index: RawData;
+  documentMapper?: DocumentMapper;
+};
+
+export const identityDocumentMapper = (doc: Document): SearchResult => {
+  return doc as unknown as SearchResult;
+};
+
+export const performSearch = async (
+  database: Orama,
+  mapper: DocumentMapper,
+  term: string
+): Promise<SearchResult[]> => {
```

Comment:
Ehm... it is already a promise. What do you mean exactly?

Reply:
You’re mixing async with Promises. I’m referring to use a pure Promise approach for this function

Reply:
Ehm... `Promise` it's just the return type. Are looking for something like this?

```
export const performSearch = (
  database: Orama,
  mapper: DocumentMapper,
  term: string
): Promise<SearchResult[]> => {
  return search(database, { term }).then(results => results.hits.map(h => mapper(h.document)));
};
```

Reply:
Yup, that’s what I meant. I also guess we don’t need a return statement as this is an one liner.

Yeah, I meant native promises rather than async/await.